papermill/hdfs.py

```diff
@@ -0,0 +1,161 @@
+# -*- coding: utf-8 -*-
+"""Utilities for working with HDFS."""
+import logging
+
+from pywebhdfs.errors import FileNotFound
+from pywebhdfs.webhdfs import PyWebHdfsClient
+
+from .utils import retry
+
+logger = logging.getLogger('papermill.hdfs')
+
+# python 2 compatibility imports
+
+try:
+    FileNotFoundError
+except NameError:
+    FileNotFoundError = IOError
+
+
+class HDFS(object):
+    """
+        :param host: the ip address or hostname of the HDFS namenode
+        :param port: the port number for WebHDFS on the namenode
+        :param user_name: WebHDFS user.name used for authentication
+        :param path_to_hosts: mapping paths to hostnames for federation
+        :param **kwargs for handling extra parameters following.
+        timeout: timeout for the underlying HTTP request
+        base_uri_pattern: format string for base URI
+        request_extra_opts: dictionary of extra options to pass
+          to the requests library (e.g., SSL, HTTP authentication, etc.)
+        >>> hdfs = PyWebHdfsClient(host='host',port='50070', user_name='hdfs')
+    """
+
+    def __init__(self, host=None, port=None, user_name=None,
+                 path_to_hosts=None, **kwargs):
+        self.__webhdfs = PyWebHdfsClient.__init__(self, host, port, user_name, **kwargs)
```

Comment:
Use single `_` for private variables, double `__` is usually for hidden system attributes.